packages/shared/createEventHook/index.ts

```diff
@@ -5,8 +5,8 @@
 import { tryOnScopeDispose } from '../tryOnScopeDispose'
 
 type Callback<T> = T extends void ? () => void : (param: T) => void
```

Comment:
[This comment](https://github.com/vueuse/vueuse/issues/3565#issuecomment-1828500148) in the original issue suggests different solution to prevent conditional type distributivity of `Callback<T>`. 

```suggestion
type Callback<T> =  [T] extends [void] ? () => void : (param: T) => void
```

Reply:
Let me check

Reply:
That works and I made some tweaks, thx~